// @bundle-stats/utils/lib/webpack/extract/assets-size-total-insight.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.extractAssetsSizeTotalInsight = void 0;
const insights_1 = require("../../config/insights");
const metrics_1 = require("../../utils/metrics");
const utils_1 = require("../utils");
const METRIC_NAME = 'totalSizeByTypeALL';
const getText = ({ metric, displayValue, displayDeltaPercentage }) => `${metric} — ${displayValue} (${displayDeltaPercentage}).`;
const extractAssetsSizeTotalInsight = (_, currentExtractedData, baselineBundleStats) => {
    const currentValue = currentExtractedData?.metrics?.[METRIC_NAME]?.value || 0;
    const baselineValue = baselineBundleStats?.metrics?.webpack?.[METRIC_NAME]?.value || 0;
    const metric = (0, utils_1.getMetricType)(METRIC_NAME);
    const info = (0, metrics_1.getMetricRunInfo)(metric, currentValue, baselineValue);
    const { displayDeltaPercentage, displayValue } = info;
    return {
        insights: {
            assetsSizeTotal: {
                type: insights_1.INSIGHT_INFO,
                data: {
                    text: getText({ metric: metric.label, displayValue, displayDeltaPercentage }),
                    md: getText({
                        metric: `*${metric.label}*`,
                        displayValue: `*${displayValue}*`,
                        displayDeltaPercentage: `*${displayDeltaPercentage}*`,
                    }),
                    info,
                },
            },
        },
    };
};
exports.extractAssetsSizeTotalInsight = extractAssetsSizeTotalInsight;
